engine/execution/state/state.go

```diff
@@ -213,17 +228,13 @@ func (s *state) NewView(commitment flow.StateCommitment) *delta.View {
 }
 
 type RegisterUpdatesHolder interface {
-	RegisterUpdates() ([]flow.RegisterID, []flow.RegisterValue)
+	RegisterUpdates() flow.RegisterEntries
 }
 
 func CommitDelta(ldg ledger.Ledger, ruh RegisterUpdatesHolder, baseState flow.StateCommitment) (flow.StateCommitment, *ledger.TrieUpdate, error) {
-	ids, values := ruh.RegisterUpdates()
+	keys, values := RegisterEntriesToKeysValues(ruh.RegisterUpdates())
```

Comment:
This is the trade-off for the safety. We have to do additional loop to create the keys and values to create the ledger update. 

But I guess we could consider changing ledger update to be a list of key-value pairs as well, instead of a list of keys and a list of values?

Reply:
This PR reduces the number of loops by 2 when compare to the original (one less loop in RegisterUpdates and one less loop in converting to ledger's keys/values)

I agree that we should convert ledger to accept key-value pair list as input as well, preferably RegisterEntries directly.  I decided against doing it in this PR to keep the PR simple.